lib/streamlit/runtime/fragment.py

```diff
@@ -119,43 +119,109 @@ def fragment(
     *,
     run_every: int | float | timedelta | str | None = None,
 ) -> Callable[[F], F] | F:
-    """Allow a function to be run independently of the full script.
-
-    Functions decorated with ``@st.experimental_fragment`` are handled specially within
-    an app: when a widget created within an invocation of the function (a fragment) is
-    interacted with, then only that fragment is rerun rather than the full streamlit app.
+    """Decorator to turn a function into a fragment which can rerun independently\
+    of the full script.
+
+    When a user interacts with an input widget created by a fragment, Streamlit
+    only reruns the fragment instead of the full script. If ``run_every`` is set,
+    Streamlit will additionally rerun the fragment at the specified interval
+    while the session is active, even if the user is not interacting with your
+    app.
+
+    To trigger a full script rerun from inside a fragment, call ``st.rerun()``
+    directly. Any values from the fragment that need to be accessed from
+    the wider app should generally be stored in Session State.
+
+    Streamlit elements created by a fragment are cleared and redrawn on each
+    fragment rerun, just like all elements are redrawn on each full-script rerun.
+    The rest of the app is persisted during a fragment rerun, with
+    fragment-generated elements retaining their relative position in the main
+    body of your app or sidebar.
+
+    Fragment code can interact with Session State, imported modules, and
+    other Streamlit elements created outside the fragment. Note that these
+    interactions are additive across multiple fragment reruns. You are
+    responsible for handling any side effects of that behavior.
 
     Parameters
     ----------
-    run_every: int, float, timedelta, str, or None
-        If set, fragments created from this function rerun periodically at the specified
-        time interval.
+    func: callable
+        The function to turn into a fragment.
```

Comment:
I was slightly unsure about this, but we do also include this parameter in [cache_data](https://github.com/streamlit/streamlit/blob/1.32.2/lib/streamlit/runtime/caching/cache_data_api.py#L409-L410), for example. So I agree with adding it here.